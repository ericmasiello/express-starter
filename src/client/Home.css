body {
  font-family: Helvetica, Arial;
  background-color: yellow;
}