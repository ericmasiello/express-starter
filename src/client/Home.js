/* @flow */
import React, { Component } from 'react';
import type { Namespace$AppState } from '../types';

export default class Home extends Component {
  constructor() {
    super();
    this.state = {
      toggle: false,
    };
  }

  state: Namespace$AppState;

  render() {
    return (
      <div>
        <h2>Home page! {this.state.toggle ? 'Yup!' : 'Nope'}</h2>
        <button onClick={() => this.setState({ toggle: !this.state.toggle })}>Toggle</button>
      </div>
    );
  }
}
