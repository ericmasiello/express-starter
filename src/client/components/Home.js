/* @flow */
import React, { Component } from 'react';
import type { Namespace$AppState } from '../../types';
import styles from './Home.scss';

export default class Home extends Component {
  constructor() {
    super();
    this.state = {
      clicks: 0,
    };
  }

  state: Namespace$AppState;

  render() {
    return (
      <div className={styles.home}>
        <button
          className={styles.button}
          onClick={() => this.setState({ clicks: this.state.clicks + 1 })}
        >
          Click to verify client side scripting: {this.state.clicks}
        </button>
        <h2>Features</h2>
        <ul>
          <li>Styling with CSS Modules, Post CSS, and Sass</li>
          <li>Flow Type Integration</li>
          <li>Express with EJS templates</li>
          <li>Babel integration for client and server</li>
          <li>React and React Router</li>
          <li>ESLint and Stylelint</li>
          <li>Logging with Bunyan</li>
        </ul>
        <h2>Dev-Mode Features</h2>
        <ul>
          <li>Hot Module Replacement</li>
        </ul>
        <h2>Production-Mode Features</h2>
        <ul>
          <li>Server Side Rendering</li>
        </ul>
      </div>
    );
  }
}
