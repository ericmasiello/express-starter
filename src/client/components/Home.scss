@import '../styles/vars';

.home {
  padding: 1.25rem;
}

.button {
  padding: 0.5rem 1rem;
  font-size: 1rem;
  border: 1px solid #ccc;
  border-radius: 3px;
  background-color: #e1e1e1;
}
