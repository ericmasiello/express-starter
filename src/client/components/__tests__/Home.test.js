import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Home from '../Home';

it('should render', () => {
  const component = renderer.create(
    <Home />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should update the counter', () => {
  const wrapper = shallow(<Home />);
  expect(wrapper.find('button').text()).toEqual('Click to verify client side scripting: 0');
  wrapper.find('button').simulate('click');
  expect(wrapper.find('button').text()).toEqual('Click to verify client side scripting: 1');
});
