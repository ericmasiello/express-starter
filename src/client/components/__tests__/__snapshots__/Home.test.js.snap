// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className={undefined}
>
  <button
    className={undefined}
    onClick={[Function]}
  >
    Click to verify client side scripting: 
    0
  </button>
  <h2>
    Features
  </h2>
  <ul>
    <li>
      Styling with CSS Modules, Post CSS, and Sass
    </li>
    <li>
      Flow Type Integration
    </li>
    <li>
      Express with EJS templates
    </li>
    <li>
      Babel integration for client and server
    </li>
    <li>
      React and React Router
    </li>
    <li>
      ESLint and Stylelint
    </li>
    <li>
      Logging with Bunyan
    </li>
  </ul>
  <h2>
    Dev-Mode Features
  </h2>
  <ul>
    <li>
      Hot Module Replacement
    </li>
  </ul>
  <h2>
    Production-Mode Features
  </h2>
  <ul>
    <li>
      Server Side Rendering
    </li>
  </ul>
</div>
`;
